test/gasless_paymaster.test.ts

```diff
@@ -13,73 +13,85 @@ import {
 } from './testutils'
 import { hexConcat, parseEther } from 'ethers/lib/utils'
 import { UserOperationStruct } from '../typechain/contracts/core/GaslessEntryPoint'
-
-describe('EntryPoint with VerifyingPaymaster', function () {
-  let entryPoint: GaslessEntryPoint
-  let walletOwner: Wallet
-  let whitelistUser: Wallet
-  let invalidUser: Wallet
-  const ethersSigner = ethers.provider.getSigner()
-
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+
+describe('Gasless EntryPoint with whitelist paymaster', function () {
+  // DummyContract.test(1, 1)
+  const dummyContractCallData = "0x4dd3b30b00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001"
+  let walletOwner: SignerWithAddress
+  let whitelistUser: SignerWithAddress 
+  let invalidUser: SignerWithAddress
   let dummyContract: DummyContract
-
+  let entryPoint: GaslessEntryPoint
   let paymaster: GaslessDemoPaymaster
+
   const fullnode: Wallet = createWalletOwner()
+
   beforeEach(async function () {
+    const [_walletOwner, _whitelistUser, _invalidUser] = await ethers.getSigners()
+    walletOwner = _walletOwner
+    whitelistUser = _whitelistUser
+    invalidUser = _invalidUser
     const DummyContract = await ethers.getContractFactory("DummyContract")
     dummyContract = await DummyContract.deploy()
     console.log(`Deploy dummy contract: ${dummyContract.address}`)
 
     entryPoint = await deployGaslessEntryPoint(fullnode.address, 1, 1)
-
-    walletOwner = createWalletOwner()
-    whitelistUser = createWalletOwner()
-    invalidUser = createWalletOwner()
-
+    //const entryPointAddr = "0xd16f6ec881e60038596c193b534c840455e66f47"
+    //const entryPoint = GaslessEntryPoint__factory.connect(entryPointAddr, walletOwner)
+    // 0xd16f6ec881e60038596c193b534c840455e66f47
+    console.log(`Deploy EntryPoint contract: ${entryPoint.address}`)
     console.log(`wallet owner: ${walletOwner.address}`)
     console.log(`User in whitelist: ${whitelistUser.address}`)
     console.log(`User not in whitelist: ${invalidUser.address}`)
 
-    paymaster = await new GaslessDemoPaymaster__factory(ethersSigner).deploy(entryPoint.address)
+    paymaster = await new GaslessDemoPaymaster__factory(walletOwner).deploy(entryPoint.address)
     console.log(`Paymaster: ${paymaster.address}`)
-    await paymaster.addStake(99999999, { value: parseEther('2') })
-    await paymaster.addWhitelistAddress(whitelistUser.address)
-    await entryPoint.depositTo(paymaster.address, { value: parseEther('1') })
   })
-    // DummyContract.test(1, 1)
-    const callData = "0x4dd3b30b00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001"
-  describe('#validatePaymasterUserOp', () => {
+  describe('#Whitelist', () => {
     it('whitelist valid', async () => {
+      await paymaster.addStake(99999999, { value: parseEther('2') })
```

Comment:
why move such operations here? let it stay in the `before_each`, every test can be reused?

Reply:
fixed